docs/proposals/dynamic-app-parameters.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Dynamic Application Parameters
+authors:
+  - "@jannfis" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all interested parties here.
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+creation-date: 2023-01-17
+last-updated: 2023-01-17
+---
+
+# Dynamic application parameters
+
+## Open questions
+
+* Should the value of a dynamic parameter be visible to a user who has `get` permissions of an Application (see also Security Implications below)?
+
+## Summary
+
+This document proposes the introduction of a new type of application parameters which are resolved dynamically at manifest rendering time. Its values will be retrieved from the application's destination cluster. Details of this are described in this proposal.
+
+## Motivation
```

Comment:
Is there a way this proposal could be generalized to also support kustomize-based applications? Such an enhancement would be similar to the `spec.postBuild.substitutionFrom` portion of the Flux Kustomization Variable Substitution feature currently missing from ArgoCD.

Reply:
That's a pretty good suggestion, although I think the post-build variable substitution is a completely different tier of feature and should be discussed in a different proposal.

I'd agree that the possibility to gather values from the cluster and use them for parameterization should not be limited to Helm, though.

Reply:
Think for example, build environment variables or parameters for plugins could directly benefit from this.